### ISSUE TITLE
Makes clear difference between files and directories

### ISSUE THREAD

**Issue by GabrielDelepine** [mihneadb/node-directory-tree#26]
Hi,

Thanks for this nice package. I have a improvement proposal and I wanted to discuss it here before making a PR :

To know if an object refer to a file or a directory, it's necessary to test if the key `extension` exists.

`extension` is the value returned by [path.extname](https://nodejs.org/api/path.html#path_path_extname_path) and if the file doesn't have extension, an empty string is return.

It means the correct condition to know if an object refer to a file is `typeof(obj.extension) === 'string'`

It's also possible to write `obj.extension !== undefined` or `obj.extension || obj.extension === ''` or `Object.keys(obj).includes('extension')` but it's not really nice.

Unfortunately, I don't think everybody will matter what happen when a file doesn't have extension and most people will probably just write `obj.extension`, witch is correct **but not for all cases**.

I'd like to add a explicit Boolean index such a `obj.isFile`. What's do you think about it ?

**Comment by mihneadb** [comment]
Hello! Actually, the way I thought you'd check for file vs folder would be the existence of the `children` key. If you think that's not enough, I'd rather suggest you added a `type` field with values `file` and `directory` (that would get stored as constants in the source code for easy use throughout).
What do you think?

**Comment by manatarms** [comment]
Hi, @GabrielDelepine 
I'm doing just that in one of my projects. Here's a snippet of my code.
```
let checkDirectory = (tree)=>{
 if (tree.children) {
     // This is a directory 
     // Do some directory processing

     // For each of it's children recurse
     tree.children.forEach(child => {
       checkDirectory(child);
     });
   } else {
     // Else it's a file
    // Do some file processing
   }
}
```
Hope this helps!

**Comment by GabrielDelepine** [comment]
Thanks for your answers

I didn't think to use `children` because I thought it could be missing in case the directory were empty. I was wrong :)

I think the proposal of @mihneadb is smart, I am going to do it. Thanks

**Comment by mihneadb** [comment]
Sounds good!

**Comment by mihneadb** [comment]
This is fixed via #28. Thanks!
